gpt_index/indices/tree/base.py

```diff
@@ -59,6 +59,7 @@ def __init__(
         num_children: int = 10,
         llm_predictor: Optional[LLMPredictor] = None,
         build_tree: bool = True,
+        async_: bool = True,
```

Comment:
imo the main use case for this actually wouldn't be building the tree index (the tree index is experimental anyways), but rather using async for response_mode="tree_summarize" during query calls. can we add an `async` parameter to the query call? e.g. in BaseGPTIndexQuery 

would be great to show a demo in the nb too

Reply:
added `use_async` to query API.